core/src/main/scala/cats/data/WriterT.scala

```diff
@@ -0,0 +1,73 @@
+package cats
+package data
+
+final case class WriterT[T[_], L, V] (run: T[(L, V)]) {
+  def written (implicit functorT: Functor[T]): T[L] =
+    functorT.map (run)(_._1)
+
+  def value (implicit functorT: Functor[T]): T[V] =
+    functorT.map (run)(_._2)
+
+  def map[Z](fn: V => Z)(implicit functorT: Functor[T]): WriterT[T, L, Z] =
+    WriterT {
+      functorT.map (run) { z =>
+        (z._1, fn (z._2))
+      }
+    }
+
+  def flatMap[U](f: V => WriterT[T, L, U])(implicit monadT: Monad[T], semigroupL: Semigroup[L]): WriterT[T, L, U] =
+    WriterT {
+      monadT.flatMap (run) { lv =>
+        monadT.map (f (lv._2).run) { lv2 =>
+          (semigroupL.combine (lv._1, lv2._1), lv2._2)
+        }
+      }
+    }
+
+  def mapValue[M, U] (f: ((L, V)) => (M, U))(implicit functorT: Functor[T]): WriterT[T, M, U] =
```

Comment:
It seems like this is more of a `mapBoth` than a `mapValue`. I think `map` already covers you if you just want to map the value component.